optimum/habana/transformers/trainer.py

```diff
@@ -992,10 +1003,12 @@ def hpu_deepspeed_checkpointing(function, *checkpoint_args, use_reentrant: Optio
                 remainder = args.gradient_accumulation_steps
             update_step = -1
             total_updates = steps_in_epoch // args.gradient_accumulation_steps + 1
+            if args.gradient_accumulation_steps == 1:
+                total_updates -= 1
             for _ in range(total_updates):
                 update_step += 1
                 num_batches = args.gradient_accumulation_steps if update_step != (total_updates - 1) else remainder
-                batch_samples, num_items_in_batch = self.get_batch_samples_transformers(epoch_iterator, num_batches)
+                batch_samples, num_items_in_batch = self.get_batch_samples(epoch_iterator, num_batches)
```

Comment:
I haven't been able to reproduce the issue. However, I noticed that others have reported similar problems https://github.com/huggingface/trl/issues/2275

what about changing that 
```py
               batch_samples, num_items_in_batch = super(Trainer, self).get_batch_samples(epoch_iterator, num_batches)
```

Reply:
That would be a better way of managing it for sure! I guess it will ultimately depend on whether or not we need to override `get_batch_samples` to avoid throughput regressions (linked to the [comment above](https://github.com/huggingface/optimum-habana/pull/1794#discussion_r1968517161)).